### DISCUSSION TITLE
DigitalOcean Budibase installation stuck on version 1.1.4

### DISCUSSION THREAD

**Discussion by xlog-null** [Budibase/budibase#8257] [Help] [answered]
I have a DigitalOcean installation which is on version 1.1.4 and I want to update it. When trying to update an app from the Budibase admin portal, it's saying it already is on the latest version:
![image](https://user-images.githubusercontent.com/92792093/195796752-c5c5a949-09f5-49ab-a9e3-d8abaf90542a.png)

How can I update the installation on a DigitalOcean droplet? It's not configured with docker-compose, it's just plain docker containers...

Thank you

**Comment by melohagan** [comment]
Hey @xlog-null 

You can update using `docker-compose pull`, the video on this page shows how you can access the docker-compose file: https://docs.budibase.com/docs/digitalocean

If you can't use docker-compose then updating would be done via: `docker pull budibase/budibase:latest`

**Comment by xlog-null** [comment]
Hi, thank you for that command - I wouldn't even know where to find it. Seems like Budibase has updated on docker: 
![image](https://user-images.githubusercontent.com/92792093/195820295-376c1884-3902-4b0d-b67e-35bab8d9ebbf.png)

However on the Budibase app managment portal, I still see that it's thinking the app version is 1.1.4: 
![image](https://user-images.githubusercontent.com/92792093/195822671-b579de17-536e-4ec3-94c9-06cf45747df9.png)
![image](https://user-images.githubusercontent.com/92792093/195822696-db5f04fb-c2d3-48cd-a25c-a4703c8128b8.png)

I've restarted the docker after update, logged out/in from Budibase. Any ideas? Thank you :)

**Reply by melohagan** [reply]
Did you stop and restart the containers as well?

**Comment by xlog-null** [comment]
Hi, after I restarted the containers, I can't connect anything. Going to an app URL gives me an error of:
![image](https://user-images.githubusercontent.com/92792093/195832102-48cd6b3b-8ec2-46b1-89d9-d8a927e50ac4.png)

And going to the main Budibase portal gives me:
![image](https://user-images.githubusercontent.com/92792093/195832173-db5b5826-2911-43d5-a7fd-38c0fd6d7992.png)

All apps are down in our company.. how to get them back up please?

**Comment by xlog-null** [comment]
Sorry, just had to restart the whole docker to get the apps back running - however, still stuck on 1.1.4...

**Comment by shogunpurple** [comment]
@xlog-null can you show the output of `docker ps`

**Comment by xlog-null** [comment, marked as answer]
@shogunpurple 

```
CONTAINER ID   IMAGE                             COMMAND                  CREATED        STATUS                             PORTS                                                                                            NAMES
692e5bdbcb8d   budibase/proxy:latest             "/docker-entrypoint.…"   2 months ago   Up 24 seconds                      80/tcp, 0.0.0.0:50000->10000/tcp, :::50000->10000/tcp                                            budibase_digitalrow_01_proxy-service_1
b91fc1771776   budibase/apps:latest              "docker-entrypoint.s…"   2 months ago   Up 25 seconds                      4001/tcp                                                                                         budibase_digitalrow_01_app-service_1
57b188ae7d2d   budibase/worker:latest            "docker-entrypoint.s…"   2 months ago   Up 24 seconds                      4001/tcp                                                                                         budibase_digitalrow_01_worker-service_1
cf2923da29ec   minio/minio                       "/usr/bin/docker-ent…"   2 months ago   Up 24 seconds (health: starting)   9000/tcp                                                                                         budibase_digitalrow_01_minio-service_1
8a26bf2345f1   ibmcom/couchdb3                   "/docker-entrypoint.…"   2 months ago   Up 24 seconds                      4369/tcp, 5984/tcp, 9100/tcp                                                                     budibase_digitalrow_01_couchdb-service_1
8d29c03dee18   redis                             "docker-entrypoint.s…"   2 months ago   Up 25 seconds                      6379/tcp                                                                                         budibase_digitalrow_01_redis-service_1
a78cbedaf81d   containrrr/watchtower             "/watchtower --debug…"   2 months ago   Up 25 seconds                      8080/tcp                                                                                         budibase_digitalrow_01_watchtower-service_1
88de71c7b093   budibase/proxy:latest             "/docker-entrypoint.…"   3 months ago   Up 24 seconds                      80/tcp, 0.0.0.0:50001->10000/tcp, :::50001->10000/tcp                                            budibase_partner-portal_01_proxy-service_1
dfdf12f4a1dd   budibase/apps:latest              "docker-entrypoint.s…"   3 months ago   Up 27 seconds                      4001/tcp                                                                                         budibase_partner-portal_01_app-service_1
bd5059415158   budibase/worker:latest            "docker-entrypoint.s…"   3 months ago   Up 24 seconds                      4001/tcp                                                                                         budibase_partner-portal_01_worker-service_1
2b0956d2f282   minio/minio                       "/usr/bin/docker-ent…"   3 months ago   Up 24 seconds (health: starting)   9000/tcp                                                                                         budibase_partner-portal_01_minio-service_1
d5fcaeb639ec   redis                             "docker-entrypoint.s…"   3 months ago   Up 25 seconds                      6379/tcp                                                                                         budibase_partner-portal_01_redis-service_1
f52f438dc924   mysql:latest                      "docker-entrypoint.s…"   3 months ago   Up 25 seconds                      0.0.0.0:3306->3306/tcp, :::3306->3306/tcp, 33060/tcp                                             mysql-01
6a7b7f313955   portainer/portainer-ee:2.12.1     "/portainer"             3 months ago   Up 26 seconds                      0.0.0.0:8000->8000/tcp, :::8000->8000/tcp, 0.0.0.0:9000->9000/tcp, :::9000->9000/tcp, 9443/tcp   portainer
8a39df1d0ad3   phpmyadmin/phpmyadmin:latest      "/docker-entrypoint.…"   4 months ago   Up 28 seconds                      0.0.0.0:50004->80/tcp, :::50004->80/tcp                                                          phpmyadmin_01
c4b58de88e5b   shlinkio/shlink:stable            "/bin/sh ./docker-en…"   4 months ago   Up 20 seconds                      0.0.0.0:50003->8080/tcp, :::50003->8080/tcp                                                      shlink_01
25a5d41df292   jc21/nginx-proxy-manager:latest   "/init"                  4 months ago   Up 24 seconds                      0.0.0.0:80-81->80-81/tcp, :::80-81->80-81/tcp, 0.0.0.0:443->443/tcp, :::443->443/tcp             nginx-proxy-manager_01
dfdb8639340b   louislam/uptime-kuma:1            "/usr/bin/dumb-init …"   4 months ago   Up 27 seconds (health: starting)   0.0.0.0:50005->3001/tcp, :::50005->3001/tcp                                                      uptime-kuma_01
5381898db88e   ibmcom/couchdb3                   "/docker-entrypoint.…"   4 months ago   Up 27 seconds                      4369/tcp, 5984/tcp, 9100/tcp                                                                     budibase_partner-portal_01_couchdb-service_1
972d762bfad1   bitnami/wordpress:latest          "/opt/bitnami/script…"   4 months ago   Up 28 seconds                      8443/tcp, 0.0.0.0:50002->8080/tcp, :::50002->8080/tcp                                            wordpress_foyer_01
d1386aeeaf1c   portainer/agent:latest            "./agent"                4 months ago   Up 24 seconds                      0.0.0.0:9001->9001/tcp, :::9001->9001/tcp                                                        portainer_agent
```

**Reply by shogunpurple** [reply]
So you said you weren't using compose, but that looks like a compose setup. 

Did you just use the standard digitalocean one click setup, or did you actually spin up each individual container and network them together with environment variables? How are you running 2 app/worker/proxy services? Are you using docker swarm or the `scale` option in compose? What commands did you run to stop and restart your containers as you said?

**Reply by shogunpurple** [reply]
Also, you can just reply to this thread instead of creating a new comment.

**Reply by xlog-null** [reply]
Before today, docker-compose wasn't even installed on this droplet. I have installed it a few hours ago trying to make some progress which was pretty much pointless as there is no compose config etc. I only literally ran install docker-compose and that's it.

This was all set up by my collegue who left without documentation. I am not sure whether he had used the one click setup or set up containers individually. I also don't know how the proxy and worker services are running, I am really new to all this. Apologies I cannot give you anymore info on this but if you point me in the right direction I can dig into it.

The command I used to restart all containers was   `docker restart $(docker ps -q)`

**Reply by shogunpurple** [reply]
I think first and foremost you need to take a backup/snapshot of your instance before you do anything here. Have you done that already?

Here's what I can see. You have 2 complete docker stacks, that have been brought up. One is running on port 50000 and the other on 50001. I'm assuming that the NGINX proxy manager container is the entry point, and routes to these 2 budibase instances. From the name of the containers in both budibase instances, it appears that they **have** been brought up with `docker compose` (notice not `docker-compose` as `compose` is now bundled in later versions of docker.) 

I think you need to find the `docker-compose.yaml` config that your colleague used, and from there, you will be able to run:
`docker compose pull && docker compose up -d` which will update all the containers. 

If your colleague has set all of this up manually and networked all the containers together without using compose, then I'm afraid that is a configuration we do not support on discussions. It's totally custom and I would need to SSH into your droplet to even see what's going on. 

You could run the image updates manually on those containers, but without the environment variables (is there an `.env` file) I cannot guarantee that it won't break anything.

I'm going to mark this one as answered, as the setup appears to be completely custom and if you find the compose file (Try this `find . -name docker-compose.yaml`) then you should be able to run the commands above. 

Thanks

**Reply by xlog-null** [reply]
I have made a backup yesterday before making any changes yes.

I cannot find the `docker-compose.yaml` file in the docker... I think it's a lost cause. I kind of want to create another droplet with the one click setup and configure it my way, if there was an easy way of transferring all apps to the new system.

Anyway, huge thanks for the help!

**Reply by shogunpurple** [reply]
You're in luck - https://docs.budibase.com/docs/backups

You can use this to migrate the old installation.

yep, to be honest without the `docker-compose.yaml`, you are kind of stuck, I don't know how you will manage and provision your installation without it. It might be easier to just migrate to another env under your control using the backups tool above.
